cheatsheets/DotNet_Security_Cheat_Sheet.md

```diff
@@ -493,13 +741,197 @@ After .NET Core 2.0 it is possible to automatically generate and verify the anti
 
 And then add the `[AutoValidateAntiforgeryToken]` attribute to the action result.
 
-## A9 Using components with known vulnerabilities
+## A7 Cross-Site Scripting (XSS)
+
+DO NOT: Trust any data the user sends you, prefer white lists (always safe) over black lists
+
+You get encoding of all HTML content with MVC3, to properly encode all content whether HTML, javascript, CSS, LDAP etc use the Microsoft AntiXSS library:
+
+`Install-Package AntiXSS`
+
+Then set in config:
+
+```xml
+<system.web>
+<httpRuntime targetFramework="4.5" 
+enableVersionHeader="false" 
+encoderType="Microsoft.Security.Application.AntiXssEncoder, AntiXssLibrary" 
+maxRequestLength="4096" />
+```
+
+DO NOT: Use the `[AllowHTML]` attribute or helper class `@Html.Raw` unless you really know that the content you are writing to the browser is safe and has been escaped properly.
+
+DO: Enable a [Content Security Policy](https://developers.google.com/web/fundamentals/security/csp/), this will prevent your pages from accessing assets it should not be able to access (e.g. a malicious script):
+
+```xml
+<system.webServer>
+    <httpProtocol>
+        <customHeaders>
+            <add name="Content-Security-Policy" 
+                value="default-src 'none'; style-src 'self'; img-src 'self'; 
+                font-src 'self'; script-src 'self'" />
+```
+
+## A8 Insecure Deserialization
+
+DO NOT: Accept Serialized Objects from Untrusted Sources
+
+DO: Prevent Deserialization of Domain Objects
+
+DO: The Serialization Process Needs to Be Encrypted So That Hostile Object Creation and Data Tampering Cannot Run
+
+DO: Run the Deserialization Code with Limited Access Permissions
+If a desterilized hostile object tries to initiate a system processes or access a resource within the server or the host’s OS, it will be denied access and a permission flag will be raised so that a system administrator is made aware of any anomalous activity on the server. 
+
+DO: Validate User Input
+Malicious users are able to use objects like cookies to insert malicious information to change user roles. In some cases, hackers are able to elevate their privileges to administrator rights by using a pre-existing or cached password hash from a previous session. 
+
+### WhiteBox Review
```

Comment:
From there this is just copy of another CS, delete it and add link to original CS

Reply:
Thanks, Did not know how to exactly reference the info